packages/verkle/src/verkleTree.ts

```diff
@@ -217,8 +217,20 @@ export class VerkleTree {
           ['put'],
         )
     }
-    // Push new/updated leafNode to putStack
-    putStack.push([leafNode.hash(), leafNode])
+    if (leafNode.values.filter((val) => val !== LeafVerkleNodeValue.Untouched).length === 0) {
```

Comment:
I was wondering, when troubleshooting over the weekend, if there should be another way to mark a node as "to be deleted" that does not require to "put untouched on all leafNode values". 

I think your intuition of a lifecycle where we "commit" and "flush" sounds right. I don't think it's necessarily what we should do now (maybe worth discussing that tomorrow on the call?) as this PR definitely brings enough improvements to be merged (and can get us to iterate on verkle tests quicker), but I also feel like we need, in the end, a better architecture.